test/qa-tests/jstests/import/mode.js

```diff
@@ -0,0 +1,88 @@
+(function() {
+  jsTest.log('Testing running import with modes');
+
+  if (typeof getToolTest === 'undefined') {
+    load('jstests/configs/plain_28.config.js');
+  }
+
+  jsTest.log('Testing running import with bad command line options');
+
+  var toolTest = getToolTest('import');
+  var db1 = toolTest.db;
+  var commonToolArgs = getCommonToolArguments();
+
+  var db = db1.getSiblingDB("upserttest");
+  db.dropDatabase();
+
+  function testWithUpsertFields(option) {
+    // This works by applying update w/ query on the fields
+    db.c.drop();
+    db.c.insert({a: 1234, b: "000000", c: 222, x: 'unique field'});
+    db.c.insert({a: 4567, b: "111111", c: 333, x: 'unique field'});
+    assert.eq(db.c.count(), 2, "collection count should be 2 at setup");
+    var ret = toolTest.runTool.apply(toolTest, ["import",
+        "--file", "jstests/import/testdata/upsert2.json",
+        option,
+        "--upsertFields", "a,c",
+        "--db", db.getName(),
+        "--collection", db.c.getName()]
+      .concat(commonToolArgs));
+
+    var doc1 = db.c.findOne({a: 1234});
+    var doc1_expect = {a: 1234, b: "blah", c: 222};
+    delete doc1["_id"];
+    if (option === "--mode=merge") {
+      doc1_expect["x"] = 'unique field';
+    }
+    assert.docEq(doc1, doc1_expect);
+
+    var doc2_1 = db.c.findOne({a: 4567, c: 333});
+    var doc2_2 = db.c.findOne({a: 4567, c: 222});
+    var doc2_1_expect = {a: 4567, b: "yyy", c: 333};
+    var doc2_2_expect = {a: 4567, b: "asdf", c: 222};
+    delete doc2_1["_id"];
+    delete doc2_2["_id"];
+    if (option === "--mode=merge") {
+      doc2_1_expect["x"] = 'unique field';
+    } else {
+      delete doc2_1["x"];
+    }
+
+    assert.docEq(doc2_1, doc2_1_expect);
+    assert.docEq(doc2_2, doc2_2_expect);
+    assert.eq(ret, 0);
+  }
+
+  function testWithoutUpsertFields(option) {
+    // This works by applying the update using _id
+    db.c.drop();
+    db.c.insert({_id: "one", a: "original value"  , x: "unique field"});
```

Comment:
The two spaces before the second comma on this line cause our JS linter to fail. Our linting settings are specified in `.eslintrc` in the root of this repo. While they do make the code look better, they should just be removed.